### PR TITLE
OpenStack: Add Kuryr information to the OpenStack README file

### PR DESCRIPTION
**Pull request by luis5tb**
This PR updates the installer information for OpenStack to include
information about how to set Kuryr SDN as well as its requirements

### CODE REVIEW

File: docs/user/openstack/README.md [hunk 1]
```diff
@@ -107,6 +153,39 @@ openstack network list --long -c ID -c Name -c "Router Type"
 
 **NOTE:** If the `neutron` `trunk` service plugin is enabled, trunk port will be created by default. for more information, please refer to [neutron trunk port](https://wiki.openstack.org/wiki/Neutron/TrunkPort).
 
+### Extra requirements when enabling Kuryr SDN
+
+
+#### Increased Quota
+
+As highlighted in the minimum quota recomentaions, when using Kuryr SDN, there
+is a need for increasing the qoutas as pods, services, namespaces, network
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: typo: quotas

File: docs/user/openstack/README.md [hunk 2]
```diff
@@ -107,6 +153,39 @@ openstack network list --long -c ID -c Name -c "Router Type"
...
+policies are using OpenStack resources. So, as an administration, the next
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: administration?

File: docs/user/openstack/README.md [hunk 3]
```diff
@@ -107,6 +153,39 @@ openstack network list --long -c ID -c Name -c "Router Type"
 
 **NOTE:** If the `neutron` `trunk` service plugin is enabled, trunk port will be created by default. for more information, please refer to [neutron trunk port](https://wiki.openstack.org/wiki/Neutron/TrunkPort).
 
+### Extra requirements when enabling Kuryr SDN
+
+
+#### Increased Quota
+
+As highlighted in the minimum quota recomentaions, when using Kuryr SDN, there
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: recomentaions?

File: docs/user/openstack/README.md [hunk 4]
```diff
@@ -107,6 +153,39 @@ openstack network list --long -c ID -c Name -c "Router Type"
 
 **NOTE:** If the `neutron` `trunk` service plugin is enabled, trunk port will be created by default. for more information, please refer to [neutron trunk port](https://wiki.openstack.org/wiki/Neutron/TrunkPort).
 
+### Extra requirements when enabling Kuryr SDN
+
+
+#### Increased Quota
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: typo: Increase

File: docs/user/openstack/README.md [hunk 5]
```diff
@@ -107,6 +153,39 @@ openstack network list --long -c ID -c Name -c "Router Type"
...
+policies are using OpenStack resources. So, as an administration, the next
+quotas should be increased for the selected project:
+
+```
+openstack quota set --secgroups 100 --secgroup-rules 500 --ports 500 --subnets 100 --networks 100 <project>
+```
+
+#### Neutron Configuration
+
+Kuryr CNI make use of the Neutron Trunks extension to plug containers into the
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: makes?

File: docs/user/openstack/README.md [hunk 6]
```diff
@@ -107,6 +153,39 @@ openstack network list --long -c ID -c Name -c "Router Type"
...
+OpenStack SDN, so the `trunks` extension must be enabled for Kuryr to properly
+work.
+
+In addition, if the default ML2/OVS Neutron driver is used, the firewall must be
+set to `openvswitch` instead of `ovs_hybrid` so that security groups are
+enforced on trunk subports and Kuryr can properly handle Network Policies.
+
+#### Octavia
+
+Kuryr SDN used Octavia OpenStack LBaaS to implement OpenShift services.
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: uses

File: docs/user/openstack/README.md [hunk 7]
```diff
@@ -107,6 +153,39 @@ openstack network list --long -c ID -c Name -c "Router Type"
...
+Thus the OpenStack enviroment must have Octavia components installled and
+configured if Kuryr SDN is used.
+
+**NOTE:** Depending on your OpenStack environment Octavia may not support UDP
+listeners, which means there is no support for UDP services if kuryr SDN is used
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: missing dot in the end of the sentence.

File: docs/user/openstack/README.md [hunk 8]
```diff
@@ -199,6 +278,64 @@ Wait for the `<cluster name>-api` server comes up and you can make your changes
 Even if the installer times out, the OpenShift cluster should still come up. Once the bootstrapping process is in place, it should all run to completion.
 So you should be able to deploy OpenShift without any floating IP addresses and DNS records and create everything yourself after the cluster is up.
 
+## Installing with Kuryr SDN
+
+To deploy with Kuryr SDN instead of the default OpenShift SDN, you simply need
+to modify the `install-config.yaml` file to include kuryr as the desired
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: Could you add kuryr with "k" uppercase, just in case someone directly copies this value from here.

File: docs/user/openstack/README.md [hunk 9]
```diff
@@ -199,6 +278,64 @@ Wait for the `<cluster name>-api` server comes up and you can make your changes
...
+
+There are known limitations when using Kuryr SDN:
+
+   * There is an amphora load balancer VM being deployed per OpenShift svc
+   with the default Octavia load balancer driver (amphora driver). If the
+   environment is resource constrained it could be a problem to create a large
+   amount of services.
+   * Depending on the Octavia OpenStack version, UDP listeners are not
+   supported. This means that OpenShift UDP services are not supported
+   * There is a known limitation of Octavia not supporting listeners on UDP
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: It might be good to highlight that only the TCP listener will be created.

File: docs/user/openstack/README.md [hunk 10]
```diff
@@ -199,6 +278,64 @@ Wait for the `<cluster name>-api` server comes up and you can make your changes
...
+### Known limitations of installing with Kuryr SDN
+
+There are known limitations when using Kuryr SDN:
+
+   * There is an amphora load balancer VM being deployed per OpenShift svc
+   with the default Octavia load balancer driver (amphora driver). If the
+   environment is resource constrained it could be a problem to create a large
+   amount of services.
+   * Depending on the Octavia OpenStack version, UDP listeners are not
+   supported. This means that OpenShift UDP services are not supported
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: missing dot in the end of sentence.

File: docs/user/openstack/README.md [hunk 11]
```diff
@@ -5,6 +5,30 @@ Support for launching clusters on OpenStack is **experimental**.
...
 
+In addition, it covers the installation with the default CNI (OpenShiftSDN),
+as well as with the Kuryr SDN
+
+## Kuryr SDN
+
+Kuryr is a CNI plug-in that uses Neutron and Octavia to provide networking
+for pods and services. It is primarily designed for OpenShift clusters that
+run on OpenStack virtual machines. Kuryr improves the network performance by
+plugging OCP pods into OpenStack SDN. In addition it provides
```
Comment (MaysaMacedo) [inline docs/user/openstack/README.md]: Missing comma: `In addition, `

File: docs/user/openstack/README.md [hunk 12]
```diff
@@ -58,6 +82,26 @@ In order to run the latest version of the installer in OpenStack, at a bare mini
      * Volume Storage: 175 Gb
      * Instances: 7
 
+#### Recommended Minimums with Kuryr SDN
```
Comment (iamemilio) [inline docs/user/openstack/README.md]:
```suggestion
### Recommended Minimums with Kuryr SDN
```

File: docs/user/openstack/README.md [hunk 13]
```diff
@@ -199,6 +279,64 @@ Wait for the `<cluster name>-api` server comes up and you can make your changes
 Even if the installer times out, the OpenShift cluster should still come up. Once the bootstrapping process is in place, it should all run to completion.
 So you should be able to deploy OpenShift without any floating IP addresses and DNS records and create everything yourself after the cluster is up.
 
+## Installing with Kuryr SDN
+
+To deploy with Kuryr SDN instead of the default OpenShift SDN, you simply need
+to modify the `install-config.yaml` file to include `Kuryr` as the desired
+`networking type` and proceed with the same steps as with the default
```
Comment (iamemilio) [inline docs/user/openstack/README.md]:
```suggestion
`networking.type` and proceed with the same steps as with the default
```

File: docs/user/openstack/README.md [hunk 14]
```diff
@@ -199,6 +279,64 @@ Wait for the `<cluster name>-api` server comes up and you can make your changes
...
+```yaml
+apiVersion: v1
+baseDomain:
+clusterID:
+controlPlane:
+  name: master
+  platform: {}
+  replicas:
+compute:
+  name: worker
```
Comment (mandre) [inline docs/user/openstack/README.md]: This needs to be a list.

Alternatively, to not have to duplicate all of the `install-config.yaml` file or have weird nil values, you could simply have:

```yaml
apiVersion: v1
...
networking:
  networkType: Kuryr
  ...
platform:
  openstack:
    ...
    trunkSupport: true
    octaviaSupport: true
    ...
```

File: docs/user/openstack/README.md [hunk 15]
```diff
@@ -199,6 +279,53 @@ Wait for the `<cluster name>-api` server comes up and you can make your changes
...
+
+### Known limitations of installing with Kuryr SDN
+
+There are known limitations when using Kuryr SDN:
+
+   * There is an amphora load balancer VM being deployed per OpenShift svc
+   with the default Octavia load balancer driver (amphora driver). If the
+   environment is resource constrained it could be a problem to create a large
+   amount of services.
+   * Depending on the Octavia OpenStack version, UDP listeners are not
```
Comment (dulek) [inline docs/user/openstack/README.md]: Just `Octavia version` or `OpenStack Octavia version`?

File: docs/user/openstack/README.md [hunk 16]
```diff
@@ -199,6 +279,53 @@ Wait for the `<cluster name>-api` server comes up and you can make your changes
...
+   supported. This means that OpenShift UDP services are not supported.
+   * There is a known limitation of Octavia not supporting listeners on UDP
+   and TCP on the same port. Thus services expose the same port for UDP and TCP
```
Comment (dulek) [inline docs/user/openstack/README.md]: It is not limited to TCP and UDP, HTTP and UDP would not work as well. I'd rephrase it to just be generic that you cannot have multiple ports using the same port.

File: docs/user/openstack/README.md [hunk 17]
```diff
@@ -199,6 +279,53 @@ Wait for the `<cluster name>-api` server comes up and you can make your changes
...
+   are not supported -- only the TCP listener will be created.
+   * Due to the above UDP limitations of Octavia, Kuryr is forcing pods to
+   use TCP for DNS resolution (`use-vc` option at `resolv.conf`). This may be a
+   problem for pods running go applications compiled with `CGO_DEBUG` flag
+   disabled as that forces to use the `go` resolver that is only using UDP and
+   is not considering the `use-vc` option added by kuryr to the `resolv.conf`.
```
Comment (dulek) [inline docs/user/openstack/README.md]: This is also an issue for [musl](https://www.musl-libc.org/)-based containers, as it's resolver doesn't support `use-vc` option. This would include e.g. images built from [alpine](https://alpinelinux.org/).
